discovery/openstack/instance.go

```diff
@@ -146,12 +211,17 @@ func (i *InstanceDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group,
 				openstackLabelUserID:         model.LabelValue(s.UserID),
 			}
 
-			flavorID, ok := s.Flavor["id"].(string)
-			if !ok {
-				level.Warn(i.logger).Log("msg", "Invalid type for flavor id, expected string")
-				continue
+			flavorName, nameOk := s.Flavor["original_name"].(string)
```

Comment:
no strong opinion but should we preserve the flavor id as the default?

Reply:
#14312

In my opinion, name is more user-friendly than ID for the users.